src/main/java/com/mraof/minestuck/entity/consort/ConsortEntity.java

```diff
@@ -402,4 +407,55 @@ public static boolean canConsortSpawnOn(EntityType<ConsortEntity> entityType, IW
 	{
 		return true;
 	}
+
+	@Override
+	public void registerControllers(AnimationData data) {
+		data.addAnimationController(createAnimation("walkAnimation", 1, this::walkAnimation));
+		data.addAnimationController(createAnimation("armsAnimation", 1, this::armsAnimation));
+		data.addAnimationController(createAnimation("deathAnimation", 1, this::deathAnimation));
+		data.addAnimationController(createAnimation("actionAnimation", 1, this::actionAnimation));
+	}
+
+	private <E extends IAnimatable> PlayState walkAnimation(AnimationEvent<E> event) {
+		if (!event.isMoving()) {
+			return PlayState.STOP;
+		}
+
+		event.getController().setAnimation(new AnimationBuilder().addAnimation("walk", true));
+		return PlayState.CONTINUE;
+	}
+
+	private <E extends IAnimatable> PlayState armsAnimation(AnimationEvent<E> event) {
+		if (!event.isMoving()) {
+			if (this.getConsortType() == EnumConsort.TURTLE) { // eeeeeeeeeeehhh maybe just fix the turtle anims instead
+				event.getController().setAnimation(new AnimationBuilder().addAnimation("armfix", true));
+				return PlayState.CONTINUE;
+			}
+			return PlayState.STOP;
+		}
+
+		event.getController().setAnimation(new AnimationBuilder().addAnimation("walkarms", true));
+		return PlayState.CONTINUE;
+	}
+
+	private <E extends IAnimatable> PlayState deathAnimation(AnimationEvent<E> event) {
+		if (dead) {
+			event.getController().setAnimation(new AnimationBuilder().addAnimation("die", false));
+			return PlayState.CONTINUE;
+		}
+		return PlayState.STOP;
+	}
```

Comment:
nakagators and salamanders currently have no death animation. Is this because Riotmode hasnt made them yet?

Reply:
precisely